### PR TITLE
chore: add Winter 25 API version, rebalance Karma tests

### PR DESCRIPTION
**Pull request by nolanlawson**
## Details

Adds API version 62 corresponding to Winter '25, and rebalances the Karma tests in CI so they run quicker overall.

This should be merged after CLCO.

## Does this pull request introduce a breaking change?

* ✅ No, it does not introduce a breaking change.

## Does this pull request introduce an observable change?

* ✅ No, it does not introduce an observable change.

### CODE REVIEW

File: packages/@lwc/shared/src/api-version.ts [hunk 1]
```diff
@@ -12,19 +12,21 @@ export const enum APIVersion {
...
+    V62_252_WINTER_25 = 62,
 }
 
 // These must be updated when the enum is updated.
 // It's a bit annoying to do have to do this manually, but this makes the file tree-shakeable,
 // passing the `verify-treeshakeable.js` test.
 
 export const LOWEST_API_VERSION = APIVersion.V58_244_SUMMER_23;
-export const HIGHEST_API_VERSION = APIVersion.V61_250_SUMMER_24;
+export const HIGHEST_API_VERSION = APIVersion.V62_252_WINTER_25;
```
Comment (wjhsf) [inline packages/@lwc/shared/src/api-version.ts:23]:
```ts
export const LOWEST_API_VERSION = allVersions.at(0)!;
export const HIGHEST_API_VERSION = allVersions.at(-1)!;
```
This still passes the tree shakeable test, but means we don't need to update every time (assuming that `allVersions` is always correctly ordered).

File: .github/workflows/karma.yml [hunk 2]
```diff
@@ -95,9 +95,45 @@ jobs:
...
+        working-directory: ./packages/@lwc/integration-karma
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v3
+
+      - name: Setup Node
+        uses: actions/setup-node@v3
+        with:
+          node-version: "18.18.0"
+          cache: 'yarn'
```
Comment (wjhsf) [inline .github/workflows/karma.yml:125]: Is having `"18.18.0"` three times enough to extract it to a variable for easier future updates? Also, mixing `"` and `'` 🥳

**Review by wjhsf — APPROVED** [review]

File: .github/workflows/karma.yml (hunk 2 above)
Comment (nolanlawson) [reply .github/workflows/karma.yml:125]: I looked into this previously, and there is no easy way to [share code](https://stackoverflow.com/a/71570780) in GitHub Actions without creating a separate repo. I can fix the mixed quotes though.

Comment (nolanlawson) [reply .github/workflows/karma.yml]: Ah maybe I can extract out just the variable though? Let me try.

File: packages/@lwc/shared/src/api-version.ts (hunk 1 above)
Comment (nolanlawson) [reply packages/@lwc/shared/src/api-version.ts:23]: Sadly we still support Safari v14, and [Array.prototype.at](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/at#browser_compatibility) is not supported there.

We could probably do this with `[length - 1]` and maybe a `PURE` comment though.

Comment (nolanlawson) [reply packages/@lwc/shared/src/api-version.ts]: Done: 9b5c4ef18

File: .github/workflows/karma.yml (hunk 2 above)
Comment (nolanlawson) [reply .github/workflows/karma.yml]: Done!

**Comment by nolanlawson** [comment]
It's post-CLCO now.

**Pull request merged**